### PR TITLE
feature: add ability to specify isolation level of transactions

### PR DESCRIPTION
**Pull request by ashleyw**
Original issue: https://github.com/odavid/typeorm-transactional-cls-hooked/issues/4

Fitting with the origins of the project, the isolation levels and descriptions were plucked from Spring Transactional.

- Upgraded `typeorm` peer dependency to `>= 0.2.8`
- Added documentation to the README (and fixed a few typos)
- Original behaviour is maintained if no `isolationLevel` is passed
- `yarn lint` passes

Thanks 👌🏼

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +48,6 @@
   "peerDependencies": {
     "cls-hooked": ">= 4.2.2",
     "reflect-metadata": ">= 0.1.12",
-    "typeorm": ">= 0.2.7"
+    "typeorm": ">= 0.2.8"
   }
-}
+}
```
Comment (odavid) [inline package.json]: Please add newline.

File: src/Transactional.ts [hunk 2]
```diff
@@ -1,29 +1,33 @@
 import { getNamespace } from 'cls-hooked'
-import { getManager } from 'typeorm'
+import { EntityManager, getManager } from 'typeorm';
```
Comment (odavid) [inline src/Transactional.ts]: Can you remove the semicolon for consistency?

**Comment by ashleyw** [comment]
Fixed 👍

**Comment by odavid** [comment]
Thanks! 👍 
Will released a new version soon

**Comment by ashleyw** [comment]
One thing I may have overlooked: if a nested method specifies a more strict isolation level, it will actually inherit the original isolation level. Do you think this is a problem?

A solution could be to check the new isolation level either matches or is less strict than the original level, and throw an error otherwise?

**Comment by odavid** [comment]
Good point: I think it is best to document the current behaviour, and explain that if a new transaction is needed with different isolation levels, `REQUIRES_NEW` is needed. Otherwise, the current transaction will be propagated.

What do you think?

**Comment by odavid** [comment]
@ashleyw - please let me know your opinion about "document only approach", before I release it.

**Comment by ashleyw** [comment]
@odavid are you thinking of just documenting in the README, or emitting warning in the code if the user attempts to propagate a non-default transaction w/o `REQUIRES_NEW`?

**Comment by odavid** [comment]
@ashleyw 😄 
I did suggest that, since I think it is a simple solution. We can always add exception logic later on.
I just don't want to release it without your approval.

**Pull request merged**